.github/workflows/main.yaml

```diff
@@ -67,57 +67,78 @@ jobs:
           CRATES_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
 
   build:
-    name: build-${{ matrix.job.target }}
-    runs-on: ${{ matrix.job.os }}
+    name: Build-${{ matrix.platform.target }}
+    runs-on: ${{ matrix.platform.os }}
     strategy:
       fail-fast: false
       matrix:
-        job:
-          - { os: ubuntu-20.04, target: aarch64-unknown-linux-gnu }
-          - { os: ubuntu-20.04, target: x86_64-unknown-linux-gnu }
-
-          - { os: macos-12, target: x86_64-apple-darwin }
-          - { os: macos-12, target: aarch64-apple-darwin }
-
-          - { os: windows-2022, target: x86_64-pc-windows-msvc }
+        platform:
+          - os_name: Linux-x86_64-gnu
+            os: ubuntu-22.04
+            target: x86_64-unknown-linux-gnu
+            bin: comtrya
+            name: comtrya-x86_64-unknown-linux-gnu
+            cross: false
+            cargo_command: cargo
+            skip_tests: false
+          - os_name: Linux-aarch64-gnu
+            os: ubuntu-22.04
+            target: aarch64-unknown-linux-gnu
+            bin: comtrya
+            name: comtrya-aarch64-unknown-linux-gnu
+            cross: false
+            cargo_command: ./cross
+            skip_tests: true
+          - os_name: Windows-x86_64
+            os: windows-2022
+            target: x86_64-pc-windows-msvc
+            bin: comtrya.exe
+            name: comtrya-x86_64-pc-windows-msvc
+            cross: false
+            cargo_command: cargo
+            skip_tests: false
+          - os_name: macOS-x86_64
+            os: macos-12
+            target: x86_64-apple-darwin
+            bin: comtrya
+            name: comtrya-x86_64-apple-darwin
+            cross: false
+            cargo_command: cargo
+            skip_tests: false
+          - os_name: macOS-aarch64
+            os: macos-12
+            target: aarch64-apple-darwin
+            bin: comtrya
+            name: comtrya-aarch64-apple-darwin
+            cross: true
+            cargo_command: ./cross
+            skip_tests: true
```

Comment:
What's the problem with the tests here?

Reply:
It may work now. I initially set them to false as I worked through the openssl dependencies and to get faster feedback on if the rest of the pipeline worked. There is a chance that I might have run into an issue with tests failing with some weird results. But we can test this out. But I remember running the aarch64 test on my x86_64 box and getting weird errors from the tests. But when I tired up an actual aarch64 Linux machine on hardware, the tests were fine? But that was also a month ago.

Reply:
Oh and looking at the original main.yaml, the only time we run tests is during the lint action, we don't actually run tests against the compiled output for each architecture
But Ill push a commit setting skip_tests to false in case we ever want to actually turn that on.